### PR TITLE
Separate test for search, and add some problem

### LINKED ISSUE #162: separate test module
Since we now have tests in most of the algorithms, I think it is a good idea to have a separate `test_module.py` file in each module. We can move all the tests from one module to it, and any further algorithms must have their tests in this file. It will make the repo easy to manage.
@keon what do you think? I can start working on this.

### PR DESCRIPTION
**Pull request by danghai**
PR for separating test for search, and add some problem.
Fix #162

### CODE REVIEW

File: search/first_occurance.py [hunk 1]
```diff
@@ -3,9 +3,6 @@
 # Approach- Binary Search
 # T(n)- O(log n)
 #
-
-import unittest
-
 def firstOccurance(array, query):
```
Comment (keon) [inline search/first_occurance.py]: first_occurance

File: search/last_occurance.py [hunk 2]
```diff
@@ -3,8 +3,6 @@
 # Approach- Binary Search
 # T(n)- O(log n)
 #
-
-
 def lastOccurance(array, query):
```
Comment (keon) [inline search/last_occurance.py]: last_occurance

File: search/test_search.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
+from binary_search import binary_search
+from first_occurance import firstOccurance
```
Comment (keon) [inline search/test_search.py]: apply name changes

**Comment by danghai** [comment]
@keon  I have fixed the name. Let me add some problem relating to search, and then merge them. 
Thanks review

**Comment by danghai** [comment]
@keon , @goswami-rahul . I am done the `search`. I already separate test for search, and add some problem relating binary search.  I also provide the `recursion technique` in binary search. Please review before merging.

File: search/findMin_rotate.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
+"""
+Suppose an array sorted in ascending order is rotated at some pivot unknown
+to you beforehand. (i.e., 0 1 2 4 5 6 7 might become 4 5 6 7 0 1 2).
+
+Find the minimum element. The complexity must be O(logN)
+
+You may assume no duplicate exists in the array.
+"""
+def findMin_rotate(array):
```
Comment (goswami-rahul) [inline search/findMin_rotate.py]: name change to `find_min_rotate` (snake_case)

File: search/findMin_rotate.py [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    while low < high:
+        mid = (low + high) // 2
+        if array[mid] > array[high]:
+            low = mid + 1
+        else:
+            high = mid
+
+    return array[low]
+
+def findMin_rotate_recur(array, low, high):
```
Comment (goswami-rahul) [inline search/findMin_rotate.py]: here too

File: search/two_sum.py [hunk 6]
```diff
@@ -0,0 +1,51 @@
+"""
```
Comment (goswami-rahul) [inline search/two_sum.py]: two_sum has already been done [here](https://github.com/keon/algorithms/blob/master/array/two_sum.py).

**Comment by danghai** [comment]
@goswami-rahul  I have updated. I know `two_sum` is already written. But `two_sum` in binary_search, It helps people think about binary search technique. It is useful too.

**Pull request merged**